src/parser/parser-options.ts

```diff
@@ -104,3 +106,24 @@ export function isTypeScript(
 
   return false;
 }
+
+/**
+ * Remove typing-related options from parser options.
+ *
+ * Allows the typescript-eslint parser to parse a file without
+ * trying to collect typing information from TypeScript.
+ *
+ * See https://typescript-eslint.io/packages/parser#withoutprojectparseroptionsparseroptions
+ */
+export function withoutProjectParserOptions(
+  options: NormalizedParserOptions,
+): NormalizedParserOptions {
+  const {
+    project: _strippedProject,
+    projectService: _strippedProjectService,
+    EXPERIMENTAL_useProjectService: _strippedExperimentalUseProjectService,
+    ...result
+  } = options;
+
+  return result;
+}
```

Comment:
I confirmed that this is same as https://github.com/typescript-eslint/typescript-eslint/blob/f30a20e083fe74d372d031d7ecafa89d2fd4fff1/packages/typescript-estree/src/withoutProjectParserOptions.ts#L9-L20